### ISSUE TITLE
Doesn't work with footnote CSL styles

### ISSUE THREAD

**Issue by alexkreidler** [timlrx/rehype-citation#14]
I used the `chicago-fullnote-bibliography.csl` provided by Zotero as follows `.use(rehypeCitation, {bibliography: "./writing/ml.bib", csl: "./csls/chicago-fullnote-bibliography.csl"})`.

The library outputs this:
```html
Default (no csl option passed): <span id="user-content-citation--wiggers_bigsciences_2022--1">(Wiggers 2022)</span>
Footnote (which chicago-fullnote csl option): <span id="user-content-citation--wiggers_bigsciences_2022--1">Kyle Wiggers, “BigScience’s AI Language Model Is Finally Available,” July 12, 2022, https://techcrunch.com/2022/07/12/a-year-in-the-making-bigsciences-ai-language-model-is-finally-available/.</span>
```
As you can see, rather than generating a numbered superscript with a link, it simply repeats the entire "fullnote" citation that should be in the footnote section.

It's rendered like this:

![Screenshot 2023-01-24 at 12-04-34 Developer Tools for Machine Learning (ML)](https://user-images.githubusercontent.com/11166947/214359688-26b477d1-5a68-4c4a-b87a-7af3de764457.png)

whereas pandoc does something like
```
<span
class="citation" data-cites="wiggers_bigsciences_2022"><a href="#fn1"
class="footnote-ref" id="fnref1"
role="doc-noteref"><sup>1</sup></a></span>
```
![Screenshot 2023-01-24 at 12-01-21 Developer Tools for Machine Learning (ML)](https://user-images.githubusercontent.com/11166947/214358701-a2de7d87-30a6-4c5a-94bf-c6bdbd3a9a2a.png)

![Screenshot 
2023-01-24 at 12-00-48 Developer Tools for Machine Learning (ML)](https://user-images.githubusercontent.com/11166947/214358501-982e96ce-be8f-4dc0-94ee-c7da97bf55bd.png)

**Comment by timlrx** [comment]
Thanks, I am able to replicate the error. Will need to have a deeper look to see why footnote styles do not work.

**Comment by timlrx** [comment]
Hi @alexkreidler, added support for fullnote and published as v0.5.0. Let me know if it works well.

**Comment by timlrx** [comment]
Great to hear that it works well! I think the second behaviour that you noted is expected and is consistent with how gfm footnotes work.
